addons/common/fnc_synchTeamColors.sqf

```diff
@@ -0,0 +1,27 @@
+/* ----------------------------------------------------------------------------
+Internal Function: CBA_fnc_synchTeamColors
+
+Description:
+    Synchs the team colors every second. Does not need to be called manually.
+
+Parameters:
+    None
+
+Returns:
+    Nothing
+
+Author:
+    BaerMitUmlaut
+---------------------------------------------------------------------------- */
+
+#include "script_component.hpp"
+
+if (leader player == player) then {
+    {
+        if ((assignedTeam _x) != (_x getVariable [QGVAR(synchedTeam), "MAIN"])) then {
+            //Local team != currently synched team, so we need to synchronize them again
+            ["CBA_teamColorChanged", [_x, assignedTeam _x]] call CBA_fnc_globalEvent;
```

Comment:
Yes, in case another player joins the squad he needs to know the assigned colors as well.